src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt

```diff
@@ -342,6 +342,12 @@ class UserService(
         updatedAt = null,
       ),
     )
+    return GetUserResponse(savedUser, true)
+  }
+
+  fun getExistingUserOrCreate(username: String): UserEntity {
+    val getUserResponse = getExistingUserOrCreate(username, dontThrowExceptionOnStaffRecordNotFound = false)
+    return getUserResponse.user!!
   }
```

Comment:
i propose we move this function definition to be above its overloaded equivalent. it may just be me but i find overloaded functions easier to understand that way?